Add 'typename' to the C++ URCU_FORCE_CAST macro so that rcu_dereference()
and related casts compile under a C++ dependent context (e.g. GCC's
-std=gnu++11 used by lttng-tools 2.14). Backported from upstream
userspace-rcu 0.15.

diff --git a/include/urcu/compiler.h b/include/urcu/compiler.h
index 2f32b38..23ee7a5 100644
--- a/include/urcu/compiler.h
+++ b/include/urcu/compiler.h
@@ -86,7 +86,7 @@
 #define __rcu
 
 #ifdef __cplusplus
-#define URCU_FORCE_CAST(_type, arg)    (reinterpret_cast<std::remove_cv<_type>::type>(arg))
+#define URCU_FORCE_CAST(_type, arg)    (reinterpret_cast<typename std::remove_cv<_type>::type>(arg))
 #else
 #define URCU_FORCE_CAST(type, arg)     ((type) (arg))
 #endif
